requests_kerberos/kerberos_.py

```diff
@@ -108,8 +111,15 @@ def generate_request_header(self, response, host, is_preemptive=False):
 
         try:
             kerb_stage = "authGSSClientInit()"
-            result, self.context[host] = kerberos.authGSSClientInit(
-                "{0}@{1}".format(self.service, host), gssflags=gssflags)
+            # contexts still need to be stored by host, but hostname_override
+            # allows use of an arbitrary hostname for the kerberos exchange
+            # (eg, in cases of aliased hosts, internal vs external, CNAMEs
+            # w/ name-based HTTP hosting)
+            kerb_host = host if not self.hostname_override else self.hostname_override
```

Comment:
Firstly, let's swap this conditional. Secondly, mind changing this to be `if not None`?